Tests/TensorFlowTests/OperatorTests/MathTests.swift

```diff
@@ -356,6 +356,48 @@ final class MathOperatorTests: XCTestCase {
         XCTAssertEqual(Double(prediction.scalars[0]), 0.816997, accuracy: 0.0001)
     }
 
+    func testQRApproximation() {
+        let shapes = [[5, 8], [3, 4, 4], [3, 3, 32, 64]]
+        for shape in shapes {
+            let a = Tensor<Float>(randomNormal: TensorShape(shape))
+            let (q, r) = a.qr()
+            let aReconstituted = matmul(q,r)
+            assertEqual(a, aReconstituted, accuracy: 1e-5)
+
+            let (qFull, rFull) = a.qr(fullMatrices: true)
+            let aReconstitutedFull = matmul(qFull, rFull)
+            assertEqual(a, aReconstitutedFull, accuracy: 1e-5)
+        }
+        
+    }
+
+    func testDiagPart() {
+        //test on 2D matrix
+        let t1 = Tensor<Float>(shape: [4, 4], scalars: (1...16).map(Float.init))
+        let target1 = Tensor<Float>([1, 6, 11, 16])
+        XCTAssertEqual(target1, t1.diagPart())
+
+        //test on 4D tensor
```

Comment:
Some minor nitpick:
- Add a space between `//` and a comment. 
- Start each comment with an upper case letter.
- End each comment with a period.
- We've been consistently adding a dash between a number and "D" when they denote dimensionality in comments.
```suggestion
        // Test on a 4-D tensor.
```